### DISCUSSION TITLE
VueUiXyCanvas not reacting to dataset updates

### DISCUSSION THREAD

**Discussion by paulh002** [graphieros/vue-data-ui#125] [Q&A] [answered]
I try to use the VueUiXyCanvas graph to display a spectrum that will update every 100 ms but the spectrum is only showing the first dataset. It are 800 data points that update every time, and the graph is very nice.   
To update the data set I use now a longpoll as test
```
const uuid = generateUUID(); 

const dataset = ref([
   {
        name: "",
        series: [1],
        type: "line",
        color: "rgb(95,139,238)",
        scaleSteps: 5,
        dataLabels: false
    }]);

async function longpolling ()
{
  try {
    const message = {"identifier": uuid, "type": "spectrum"};
    const response = await axios.post('/api/spectrum', message);
    //console.log('Fetched data:', response.data); // Log the fetched data to the console
    dataset.value[0].series = response.data;
    //console.log("data",dataset.value[0].series); 
  }
    catch (error) {
    console.error('error fetching wsjtx data', error);
  }
  finally {
    setTimeout(longpolling, 1);
  }
}

onMounted(async () => {
  longpolling();
});
```

If I replace the component with the normal XY component the update works. But the response of this component is to slow. So I was hoping the Canvas version will be faster.

**Comment by graphieros** [comment, marked as answer]
Hi Paul :)

Thanks for pointing this out, a watcher on the dataset was missing.
You can upgrade to v2.4.22 which solves this issue.

Using VueUiXyCanvas is a good choice indeed for this kind of big dataset with frequent update, as the browser doesn't have to manage thousands of DOM nodes.

Cheers

**Comment by paulh002** [comment]
Yes, now it works fine. Thanks!

**Comment by paulh002** [comment]
Hi, I have some questions
Is there a way to get an indicator? Also to have data labels on top of the graph. Also the time data labels need to match the amount of datapoints even when you use a modula. Or should I use slots to draw custom data labels? I want to replicate the graph which I can get on my embedded device, 

![image](https://github.com/user-attachments/assets/f525058c-17a9-4f13-845d-2a57e1a30b12)

**Reply by graphieros** [reply]
To display time labels on top, you need to tweak `style.chart.grid.y.timeLabels.offsetY` with a negative value. You might need to tweak `style.chart.paddingProportions` too, to give some space for top labels.

For the indicator, you can tweak `style.chart.selector`.

I'm not sure I understand your question regarding the time labels.
What did you put inside `style.chart.grid.y.timeLabels.values` ?

**Comment by paulh002** [comment]
Ok, I will try. The problem with time labels is that it looks like I have to add 800 time labels and use modula to select the subset I want to see. I was wondering If I could only add the time labels I want to display. So only 9 time lables.

**Reply by graphieros** [reply]
Having 800 time labels set allows to show the label in the tooltip, or when using the selector, on any datapoint.

https://github.com/user-attachments/assets/32e1b3e1-517a-4a7a-a810-0bd521b967ac

**Comment by paulh002** [comment]
Ok, I will try that. Thanks for you help.

**Reply by graphieros** [reply]
You are welcome, thanks for trying out the lib :)

**Comment by paulh002** [comment]
I forgot one question. Is it possible to fix the scale of the chart? It looks like the scale is depending on the largest y value. But in realtime you want to have a fixed y scale.

**Reply by graphieros** [reply]
Yes, I will add config attributes to set scaleMin and scaleMax values.
Will try to do that after work today.

**Comment by paulh002** [comment]
Thanks! that would be great.

**Reply by graphieros** [reply]
@paulh002 

You can upgrade to [v2.4.24](https://github.com/graphieros/vue-data-ui/releases/tag/v2.4.24) which implements custom scales on VueUiXyCanvas.

For your use case, I think you'll only have to set your scales in the config:

```ts
style.chart.scale.min: number | null;
style.chart.scale.max: number | null;
```

**Comment by paulh002** [comment]
Thanks! will test it

**Comment by paulh002** [comment]
Perfect
![image](https://github.com/user-attachments/assets/3244a835-b641-4a8b-a21c-b1a454ebbbea)

**Comment by paulh002** [comment]
One other observation, it looks like the vertical lines are not being displayed. Horizontal lines do work.

**Reply by graphieros** [reply]
Yes, they are disabled above a certain amount of datapoints.
You need the same modulo as time labels right ?

**Comment by paulh002** [comment]
yes :)

**Reply by graphieros** [reply]
Ok, lemme check it out

**Comment by graphieros** [comment]
@paulh002 
You can upgrade to v2.4.25

If you already set `style.chart.grid.y.verticalLines.show:true` you should be good to go.

**Comment by paulh002** [comment]
Yes, thanks
![image](https://github.com/user-attachments/assets/2b26908a-ac66-4660-bd77-e87494254f3d)
